components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/model/ActionExecutionStatus.java

```diff
@@ -28,14 +28,26 @@
 public class ActionExecutionStatus {
 
     private final Status status;
-    private final Map<String, Object> responseContext;
+    private Map<String, Object> responseContext;
+    private Response response;
+
+    public ActionExecutionStatus(Status status) {
+
+        this.status = status;
+    }
 
     public ActionExecutionStatus(Status status, Map<String, Object> responseContext) {
 
         this.status = status;
         this.responseContext = responseContext;
     }
 
+    public ActionExecutionStatus(Status status, Response response) {
+
+        this.status = status;
+        this.response = response;
+    }
+
```

Comment:
If you have introduced this constructor why you need to populate the eventcontext map with error info ?